internal/store/interfaces.go

```diff
@@ -58,3 +58,27 @@ type Store interface {
 	UpsertRoute(ctx context.Context, route Route, updateConditionFn func(current Route) bool) error
 	Sync(ctx context.Context) error
 }
+
+// Backend is used for persisting and querying gateways and routes
+type Backend interface {
+	GetGateway(ctx context.Context, id core.GatewayID) ([]byte, error)
+	ListGateways(ctx context.Context) ([][]byte, error)
```

Comment:
@andrewstucki we're either going to have to make this list function more robust to support filtering to one or more namespaces, or we'll have to list all and filter further up in the application layer. Thoughts?

Reply:
Not entirely sure yet, for the first pass it's probably just easier to filter at a higher level, but I'm thinking we may want to move that logic into the actual store retrieval mechanism at some point to avoid the hit of the roundtrip/compute cost for Consul persistence.

Reply:
We'll leave as-is for now and revisit in the coming weeks